Source/Urho3D/Core/ProcessUtils.cpp

```diff
@@ -487,74 +414,12 @@ ea::string GetPlatformName()
 
 unsigned GetNumPhysicalCPUs()
 {
-#if defined(UWP)
-    // This is as good as it gets on UWP
-    return std::thread::hardware_concurrency();
-#elif defined(IOS)
-    host_basic_info_data_t data;
-    GetCPUData(&data);
-#if TARGET_OS_SIMULATOR
-    // Hardcoded to dual-core on simulator mode even if the host has more
-    return Min(2, data.physical_cpu);
-#else
-    return data.physical_cpu;
-#endif
-#elif defined(TVOS)
-#if TARGET_OS_SIMULATOR
-    return Min(2, SDL_TVOS_GetActiveProcessorCount());
-#else
-    return SDL_TVOS_GetActiveProcessorCount();
-#endif
-#elif defined(__linux__)
-    struct CpuCoreCount data{};
-    GetCPUData(&data);
-    return data.numPhysicalCores_;
-#elif defined(__EMSCRIPTEN__)
-#ifdef __EMSCRIPTEN_PTHREADS__
-    return emscripten_num_logical_cores();
-#else
-    return 1; // Targeting a single-threaded Emscripten build.
-#endif
-#else
-    struct cpu_id_t data;
-    GetCPUData(&data);
-    return (unsigned)data.num_cores;
-#endif
+    return ea::max(1u, std::thread::hardware_concurrency());
```

Comment:
Even though this function was not always reliable i think it would be worthwhile to slap a `[[deperecated]]` here and add a warning log to discourage it's use and stick behavior change to everyone's face.